.github/workflows/pytest.yml

```diff
@@ -57,7 +57,7 @@ jobs:
         id: coverage
         run: |
           pip install coverage pytest pytest-random-order pytest-cov
-          pytest --cov --random-order --cov-branch --cov-report=xml tests/unit_tests/
+          pytest --cov --random-order --cov-branch --cov-report=xml --maxfail=2 --no-cov-on-fail tests/unit_tests/
```

Comment:
not sure about the maxfail. While it does make sense for long-running jobs like e2e tests, our unit tests are probably still sufficiently fast.

Reply:
ah yeah good point, fixed